### PR TITLE
IBX-1136: Prevent legacy and symfony clear:cache script from clearing SPI cache

### PR DESCRIPTION
**Pull request by vidarl**
JIRA ticket: https://issues.ibexa.co/browse/IBX-1136

Make it configurable if
- clearing legacy cache ( view-cache) should also clear SPI cache
- running `bin/console cache:clear` should also clear SPI cache

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -118,3 +118,23 @@ rewrite "^/var/storage/packages/(.*)" "/var/storage/packages/$1" break;
...
+    clear_all_spi_cache_from_legacy: false
+```
+
+Legacy bridge will also alter the default behaviour of Symfony's `bin/console cache:clear` command so that SPI cache
+is cleared when it is executed. This behaviour can be disabled using the setting:
+
+```
+// app/config/ezplatform.yml
+ez_publish_legacy:
+    clear_all_spi_cache_on_symfony_clear_cache: true
```
Comment (gggeek) [inline INSTALL.md:139]: should be false ?

Comment (alongosz) [reply INSTALL.md:139]: > should be false ?

No, that would be a BC break.

File: INSTALL.md [hunk 2]
```diff
@@ -118,3 +118,23 @@ rewrite "^/var/storage/packages/(.*)" "/var/storage/packages/$1" break;
 Last step, if you are on *nix operation system, is to make sure to run
 the appropriate command for setting correct folder permissions, you
 can find the information you need in [installation guide for eZ Platform](https://doc.ezplatform.com/en/latest/getting_started/install_ez_platform/).
+
+### Clearing of cache
+
+Legacy bridge will by default also clear SPI cache ( stored in redis, memcached or disk) when content-view cache is
```
Comment (glye) [inline INSTALL.md:124]:
```suggestion
Legacy bridge will by default also clear SPI cache (stored in redis, memcached or disk) when content-view cache is
```

File: INSTALL.md (hunk 1 above)
Comment (Steveb-p) [reply INSTALL.md:139]: Aye @gggeek, you're actually right.

**Review by glye — APPROVED** [review]

File: bundle/Cache/PersistenceCachePurger.php [hunk 3]
```diff
@@ -73,13 +76,23 @@ public function __construct(
...
+        }
+    }
+
+    /**
+     * Clear all persistence cache.
      *
      * Sets a internal flag 'allCleared' to avoid clearing cache several times
      */
-    public function all()
+    protected function flushSpiCache()
```
Comment (Steveb-p) [inline bundle/Cache/PersistenceCachePurger.php:95]: We don't need this method to be exposed at all.

```suggestion
    private function flushSpiCache()
```

File: bundle/Resources/config/services.yml [hunk 4]
```diff
@@ -222,6 +222,8 @@ services:
     ezpublish_legacy.persistence_cache_purger:
         class: eZ\Bundle\EzPublishLegacyBundle\Cache\PersistenceCachePurger
         public: false
+        calls:
+            - [setContainer, ["@service_container"]]
         arguments:
             - "@ezpublish.cache_pool"
             - "@ezpublish.spi.persistence.cache.locationHandler"
```
Comment (Steveb-p) [inline bundle/Resources/config/services.yml:229]:
```suggestion
            - "@ezpublish.spi.persistence.cache.locationHandler"
            - "%ezpublish_legacy.clear_all_spi_cache_on_symfony_clear_cache%"
            - "%ezpublish_legacy.clear_all_spi_cache_from_legacy%"
```

**Review by alongosz — CHANGES_REQUESTED** [review]
@vidarl please refactor this to inject parameters instead, like discussed during internal sync.

File: bundle/Resources/config/services.yml (hunk 4 above)
Comment (vidarl) [reply bundle/Resources/config/services.yml:227]: Fixed in d6e4d9d36fe2ee9291967ef14598f3aa3e83690e

**Review by alongosz — APPROVED** [review]

File: bundle/Cache/PersistenceCachePurger.php [hunk 5]
```diff
@@ -65,21 +75,35 @@ class PersistenceCachePurger implements CacheClearerInterface
...
-        CacheIdentifierGeneratorInterface $cacheIdentifierGenerator
+        CacheIdentifierGeneratorInterface $cacheIdentifierGenerator,
+        bool $clearAllSPICacheOnSymfonyClearCache = true,
+        bool $clearAllSpiCacheFromLegacy = true
     ) {
         $this->cache = $cache;
         $this->locationHandler = $locationHandler;
         $this->cacheIdentifierGenerator = $cacheIdentifierGenerator;
+        $this->clearAllSPICacheOnSymfonyClearCache = $clearAllSPICacheOnSymfonyClearCache;
+        $this->clearAllSpiCacheFromLegacy = $clearAllSpiCacheFromLegacy;
```
Comment (Steveb-p) [inline bundle/Cache/PersistenceCachePurger.php:86]: Property / variable naming is inconsistent.

**Review by Steveb-p — APPROVED** [review]

File: bundle/Cache/PersistenceCachePurger.php (hunk 5 above)
Comment (vidarl) [reply bundle/Cache/PersistenceCachePurger.php]: Fixed

**Comment by micszo** [comment]
@vidarl don't we need a doc update for this? Is INSTALL.md enough?

**Comment by vidarl** [comment]
> @vidarl don't we need a doc update for this? Is INSTALL.md enough?
It seems like the documentation we had on legacy bridge has been removed from doc ( AFAIK, we at least had a doc covering how to install it ). We only refer to the .md now AFAIK : https://doc.ibexa.co/en/latest/migrating/migrating_from_ez_publish_platform/#25-optional-install-legacy-bridge

@DominikaK

**Comment by DominikaK** [comment]
We never had any doc relating to Bridge, with the exception of migration guides.
Anyways, is Bridge supported with any of the currently supported versions?

**Comment by micszo** [comment]
It is with 2.5, we have such a customer.

**Comment by vidarl** [comment]
Just add 427eff8d7fc0d744fdab47304849597b1136e727, A clear-all from legacy would also issue clearing ContentTypes ( type_map) from persistance cache. ( Discovered by @micszo )

**Review by micszo — APPROVED** [review]
QA Approved on eZ Platform EE 2.5 with Legacy Bridge with Redis with diff.

**Comment by micszo** [comment]
@vidarl @glye not sure who's the Maintainer here but I think it's ready to be merged.

**Comment by lserwatka** [comment]
@glye you can proceed with merging and release.

**Pull request merged**